### PR TITLE
feat: add event properties, events, and did hash

### PR DESCRIPTION
**Pull request by zachferland**

### CODE REVIEW

**Comment by zachferland** [comment]
not done yet :)  adding did hash property to events in this PR as well

**Comment by zachferland** [comment]
Adds request origin to api calls (so api events can be filtered by domain/app)
Adds did_hash property to node events (besides thread update)
Adds thread name and thread space to update thread events 
Adds flag for opendb/pindb event to indicate whether new account or not
Adds event for priv store update 

cc @dazuck

**Comment by zachferland** [comment]
All set now

File: src/pinning.js [hunk 1]
```diff
@@ -285,41 +287,82 @@ class Pinning {
...
+            conf.links.push(obj)
+          }
+          return conf
+        }, Promise.resolve({}))
+
+      const link = config.links[0]
+      const did = /\bdid:.*\b/g.exec(link.message)[0]
+      return did
+    } catch (e) {
+      return null
```
[the remark this answers is not in the document]
Comment (zachferland) [reply src/pinning.js]: yeah, was easier than getting from jwt in pub store, but actually yeah Ill add an pin db event for earlier lib users, just won't have the did_hash

File: src/pinning.js [hunk 2]
```diff
@@ -285,41 +287,82 @@ class Pinning {
...
     tick.stop()
   }
 
-  async rewriteDBCache (odbAddress, rootStoreAddress) {
+  async rootStoreToDID (rootStoreAddress) {
+    try {
+      const config = await this.openDBs[rootStoreAddress].db
+        .iterator({ limit: -1 })
+        .collect()
+        .reduce(async (conf, entry) => {
```
[the remark this answers is not in the document]
Comment (zachferland) [reply src/pinning.js]: yeah could change that now 👍

File: src/pinning.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (oed) [reply src/pinning.js]: ok, makes sense 👍

**Comment by zachferland** [comment]
did_hash moved to event id
add redundant event for space update with space as id, so we can get uniques in MP
add pin event for rootstores with no links
switch to find

**Pull request merged**